corehq/util/workbook_reading/adapters/csv.py

```diff
@@ -0,0 +1,125 @@
+import ast
+from contextlib import contextmanager
+import datetime
+from dateutil.parser import parse
+import csv
+import os
+
+from corehq.util.workbook_reading import (
+    SpreadsheetFileEncrypted,
+    SpreadsheetFileInvalidError,
+    SpreadsheetFileNotFound,
+    Workbook,
+)
+from .raw_data import make_worksheet
+
+
+@contextmanager
+def open_csv_workbook(filename):
+    try:
+        if os.stat(filename).st_size <= 1:
+            raise SpreadsheetFileInvalidError('File is empty')
+        with open(filename, "r") as csv_file:
+            yield _CSVWorkbookAdaptor(csv_file).to_workbook()
+    except UnicodeDecodeError as error:
+        raise SpreadsheetFileEncrypted(str(error))
+    except csv.Error as error:
+        if str(error) == 'line contains NUL':
+            raise SpreadsheetFileEncrypted('Workbook is encrypted')
+        else:
+            raise SpreadsheetFileInvalidError(str(error))
+    except IOError as error:
+        raise SpreadsheetFileNotFound(str(error))
+    except FileNotFoundError as error:
+        raise SpreadsheetFileNotFound(str(error))
+
+
+def format_value(str_value):
```

Comment:
Would be great to have unit tests for this function. Something along the lines of [test_excel_format_value.py](https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/couchexport/tests/test_excel_format_value.py)